### PR TITLE
SimpleNLPModel test

### PR DESCRIPTION
**Pull request by abelsiqueira**
Not working on 0.5.
Uses `SimpleNLPModel` from JuliaOptimizers/NLPModels.jl#4.

### CODE REVIEW

**Comment by dpo** [comment]
@abelsiqueira I'm so sorry. Somehow I didn't get notified of this PR and I just pushed build scripts for Travis and Appveyor. I didn't mean to ignore your contribution! I like your test on `SimpleNLPModel`. Do you want to rebase and add it?

Sorry again!

**Comment by abelsiqueira** [comment]
No problem. I'll make a slight modification on travis to separate some steps,
and to use my NLPModels while you don't merge JuliaOptimizers/NLPModels.jl#4

File: REQUIRE [hunk 1]
```diff
@@ -0,0 +1,2 @@
+JuMP
```
Comment (dpo) [inline REQUIRE]: Technically, `JuMP` isn't required. We should be finer here.

File: REQUIRE [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+DistributedArrays
```
Comment (dpo) [inline REQUIRE]: This should really go into `Krylov.jl`. Perhaps I should stop supporting Julia 0.3.

File: .travis.yml [hunk 3]
```diff
@@ -30,10 +28,15 @@ before_install:
...
+
+install:
+  - julia -e 'versioninfo()'
+  - julia -E 'Pkg.clone(pwd())'
+  - julia -E 'Pkg.build("Optimize")'
 
 script:
-  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
-  - julia -e 'Pkg.clone(pwd()); Pkg.build("Optimize"); Pkg.test("Optimize"; coverage=true)'
+  - julia --check-bounds=yes -E 'Pkg.test("Optimize", coverage=true)'
```
Comment (dpo) [inline .travis.yml]: Great!

File: test/runtests.jl [hunk 4]
```diff
@@ -2,6 +2,17 @@ using Optimize
...
+g(x) = x
+H(x) = eye(2)
+
+nlp_simple = SimpleNLPModel(x0, f, grad=g, hess=H)
+
+for nlp in [nlp_jump; nlp_simple]
+  for solver in [:trunk, :lbfgs]
+    stats = run_solver(solver, nlp, verbose=true)
+  end
+end
```
Comment (dpo) [inline test/runtests.jl]: Ok. In time we'll need actual tests, that `assert` or `@test` stuff. But this is fine for now.

Comment (abelsiqueira) [inline test/runtests.jl]: I may start with TRON as part of the workflow.

**Comment by dpo** [comment]
Does this need to be revised?

**Comment by abelsiqueira** [comment]
Yes, all Optimize work is stopped while the other packages were being updated. I'm afraid I did a depth first work here.

**Comment by dpo** [comment]
I can't tell where this stands.

**Comment by abelsiqueira** [comment]
Let me take a look.

**Comment by abelsiqueira** [comment]
This is all obsolete.

**Pull request closed without merging**